File: sdk-node/src/propagation.ts
export interface TraceContext {
  traceId: string
  spanId: string
}

export function injectHeaders(ctx: TraceContext, headers: Record<string, string>): void {
  headers['traceparent'] = `00-${ctx.traceId.replace('trace_', '').padStart(32, '0')}-${ctx.spanId.replace('span_', '').padStart(16, '0')}-01`
  headers['x-traceflow-trace-id'] = ctx.traceId
  headers['x-traceflow-span-id'] = ctx.spanId
}

export function extractContext(headers: Record<string, string | string[] | undefined>): TraceContext | null {
  const xTraceId = header(headers, 'x-traceflow-trace-id')
  const xSpanId = header(headers, 'x-traceflow-span-id')
  if (xTraceId && xSpanId) {
    return { traceId: xTraceId, spanId: xSpanId }
  }

  const traceparent = header(headers, 'traceparent')
  if (traceparent) {
    const parts = traceparent.split('-')
    if (parts.length === 4) {
      return {
        traceId: 'trace_' + parts[1].replace(/^0+/, ''),
        spanId: 'span_' + parts[2].replace(/^0+/, ''),
      }
    }
  }

  return null
}

function header(headers: Record<string, string | string[] | undefined>, key: string): string | undefined {
  const val = headers[key] ?? headers[key.toLowerCase()]
  return Array.isArray(val) ? val[0] : val
}

File: sdk-python/traceflow/propagation.py
def extract_context(headers: dict):
    trace_id = None
    span_id = None
    
    # Try traceparent first (W3C)
    traceparent = headers.get("traceparent")
    if traceparent:
        parts = traceparent.split("-")
        if len(parts) == 4:
            trace_id = parts[1]
            span_id = parts[2]
            return {"traceId": trace_id, "spanId": span_id}
            
    # Try custom headers
    trace_id = headers.get("x-traceflow-trace-id") or headers.get("x-b3-traceid")
    span_id = headers.get("x-traceflow-span-id") or headers.get("x-b3-spanid")
    
    if trace_id:
        return {"traceId": trace_id, "spanId": span_id}
        
    return None

def inject_headers(span, headers: dict):
    headers["x-traceflow-trace-id"] = span.trace_id
    headers["x-traceflow-span-id"] = span.id
    headers["traceparent"] = f"00-{span.trace_id}-{span.id}-01"
